homeassistant/components/zwave_js/__init__.py

```diff
@@ -131,13 +131,15 @@ def async_on_node_removed(node: ZwaveNode) -> None:
         # grab device in device registry attached to this node
         dev_id = get_device_id(client, node)
         device = dev_reg.async_get_device({dev_id})
-        # note: removal of entity registry is handled by core
+        assert device
+        # note: removal of entity registry entry is handled by core
         dev_reg.async_remove_device(device.id)
 
     @callback
     def async_on_value_notification(notification: ValueNotification) -> None:
         """Relay stateless value notification events from Z-Wave nodes to hass."""
         device = dev_reg.async_get_device({get_device_id(client, notification.node)})
+        assert device
```

Comment:
```suggestion
        if not device:
            return
```

Reply:
Modified to explicitly silence the MyPy errors for now, let's do proper error handling + tests in a separate PR